src/AppInstallerCLICore/Workflows/PortableInstallFlow.cpp

```diff
@@ -0,0 +1,376 @@
+                context.Get<Execution::Data::Installer>()->Arch,
+                productCode);
+
+            if(uninstallEntry.Exists())
+            {
+                if (uninstallEntry.IsSamePortablePackageEntry(packageIdentifier, sourceIdentifier))
+                {
+                    // TODO: Replace HashOverride with --Force when argument behavior gets updated.
+                    if (!context.Args.Contains(Execution::Args::Type::HashOverride))
+                    {
+                        AICLI_LOG(CLI, Error, << "Registry match failed, skipping write to uninstall registry");
+                        AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_PORTABLE_PACKAGE_ALREADY_EXISTS);
+                    }
+                    else
+                    {
+                        AICLI_LOG(CLI, Info, << "Overriding registry match check...");
+                        context.Reporter.Warn() << Resource::String::PortableRegistryCollisionOverridden << std::endl;
+                    }
+                }
+            }
+
+            AICLI_LOG(CLI, Info, << "Begin writing to Uninstall registry.");
+            uninstallEntry.SetValue(PortableValueName::DisplayName, entry.DisplayName);
+            uninstallEntry.SetValue(PortableValueName::DisplayVersion, entry.DisplayVersion);
+            uninstallEntry.SetValue(PortableValueName::Publisher, entry.Publisher);
+            uninstallEntry.SetValue(PortableValueName::InstallDate, Utility::GetCurrentDateForARP());
+            uninstallEntry.SetValue(PortableValueName::URLInfoAbout, manifest.DefaultLocalization.Get<Manifest::Localization::PackageUrl>());
+            uninstallEntry.SetValue(PortableValueName::HelpLink, manifest.DefaultLocalization.Get<Manifest::Localization::PublisherSupportUrl>());
+            uninstallEntry.SetValue(PortableValueName::UninstallString, L"winget uninstall --product-code " + productCode);
+            uninstallEntry.SetValue(PortableValueName::WinGetInstallerType, ConvertToUTF16(InstallerTypeToString(InstallerTypeEnum::Portable)));
+            uninstallEntry.SetValue(PortableValueName::WinGetPackageIdentifier, manifest.Id);
+            uninstallEntry.SetValue(PortableValueName::WinGetSourceIdentifier, sourceIdentifier);
+            uninstallEntry.SetValue(PortableValueName::PortableTargetFullPath, GetPortableTargetFullPath(context).wstring());
+            uninstallEntry.SetValue(PortableValueName::PortableSymlinkFullPath, GetPortableSymlinkFullPath(context).wstring());
+            uninstallEntry.SetValue(PortableValueName::SHA256, Utility::SHA256::ConvertToWideString(context.Get<Execution::Data::HashPair>().second));
+            uninstallEntry.SetValue(PortableValueName::InstallLocation, GetPortableTargetDirectory(context).wstring());
+            AICLI_LOG(CLI, Info, << "Writing to Uninstall registry complete.");
+        }
+
+        void MovePortableExeAndCreateSymlink(Execution::Context& context)
```

Comment:
I don't know what it is, but something about this name just doesn't seem right. I think it is the `And` which indicates that this task is doing two tasks and that it may be best to split it into one task for moving the exe, and one for creating the symlink? I'm not an expert though, so I'd like others' opinions on this too

cc @JohnMcPMS

Reply:
I agree that it would generally make more sense to split the two operations if they are logically independent.  I don't think that we need to now, but I definitely foresee the need for this in the future for portable(s) in an archive support, where it isn't just a single file and symlink.  I think it can wait until then though.

As for the name, again I'm fine with it for now because I expect it to be refactored for the archive support.